### PR TITLE
Improve number formatting

### LINKED ISSUE #3733: Improving Number Formatting
### Description

Since we have been using Numeral formatting in the DApp for awhile now, we have had a chance to see how it works in different cases and also receive some feedback on how numbers are presented. The following is a list of number formatting rules we would like to implement.

1. Abbreviate only above 1 million, i.e. we should not see thousands abbreviated anywhere like the screenshot [IMAGE 1].
2. Anything over 1,000 should not have more then 2 decimal places.
3. Numbers below 1 can have up to 5 decimals.
4.  Ensure consistency with spacing for e.g. [IMAGE 2] should have a space.
5. The main Colony balance should not be abbreviated
6. All abbreviated numbers should show the full number on hover, like addresses, or using a tooltip.
7. Use engineering notation above 1 trillion and numbers less then 0.00001

IMAGE 1:

download (67)

IMAGE 2: 

Screenshot 2022-08-09 202820

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

This PR attempts to refactor number formatting in the dapp following the rules outlined in https://www.notion.so/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb.

Looking forward to hearing your thoughts on the solution.

## Changes
- Added types for `move-decimal-point` library
- Refactored `Numeral` component:
  - It now follows the rules as described in the doc
  - It got simplified by removing the appearance prop (actually, most of the consuming components have been overriding it with own custom styles) and the ability to override the formatting rules (it wasn't used anywhere)
- Edited the consuming components to work with the new `Numeral` component

## Deletions
- Removed unused `PayoutsList` component

## TODO

- [x] Ensure consistency with spacing 
- [x] New Numeral component
- [x] Switch everything to use the new Numeral component

Resolves #3733

### CODE REVIEW

**Comment by willm30** [comment]
Don't forget to select "JoinColony/frontend" from the reviewers list so we get a notification when you're ready for a review 👍

**Review by willm30 — COMMENTED** [review]
Nice work Jakub! I've been looking for inconsistencies with [the proposal](https://www.notion.so/colony/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb) but only found one thing I wanted to query, which is the treatment of large billions. 

999,999,999,999 was rounded to 1T, and 999,000,000,000 was 0.999T, which I don't see in the proposal. I was expecting 999.999B and 999B respectively. 

Also, with respect to tiny numbers (0.00000...), the tooltip is not currently implemented on the action page, which I think it should be? I actually like the idea of engineering notation here, as at a glance I think it would be more informative than a string of zeroes. Then we could also have a tooltip as well, if the user wants to see the original value. I mention since you raised the question in the proposal, so I'd be interested to see how it looks. 

Final couple of things. There are a few build errors in your second commit re the omission of BigNumber type from the type definition of `moveDecimal`. 

And my instinct with regard to the removal of the appearance prop is that it should be left in and continued to be overriden where necessary. I say this only because I expect it was designed with that in mind originally, and follows the same custom as other core components in the dapp. But I might be mistaken here, so worth a second opinion.

File: src/modules/core/components/Numeral/NewNumeral.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+interface PrefixSuffixOptions {
+  prefix?: string;
+  suffix?: string;
+}
+
+export const getValueWithPrefixAndSuffix = (
+  value: string,
+  { prefix, suffix }: PrefixSuffixOptions,
+) => {
+  return `${prefix ? `${prefix} ` : ''}${value}${suffix ? ` ${suffix}` : ''}`;
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/modules/core/components/Numeral/NewNumeral.tsx:36]: That's what I initially did as well, turns out we have a rule against it:
image
I'm not sure it's worth adding comments to disable this for a minor improvement (if any) in readability

Noted your point on removing the unnecessary brackets and return statement, thanks!

File: src/modules/core/components/Numeral/NewNumeral.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/modules/core/components/Numeral/NewNumeral.tsx:1]: Good spot!

File: src/modules/core/components/Numeral/NewNumeral.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  unit?: string;
+
+  prefix?: string;
+
+  suffix?: string;
+
+  className?: string;
+}
+
+const NewNumeral = ({
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/modules/core/components/Numeral/NewNumeral.tsx:55]: Removed this file but I was also missing it in other components - fixed

File: src/modules/core/components/Numeral/NewNumeral.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/core/components/Numeral/NewNumeral.tsx]: Fair enough 👍

**Comment by jakubcolony** [comment]
>999,999,999,999 was rounded to 1T, and 999,000,000,000 was 0.999T, which I don't see in the proposal. I was expecting 999.999B and 999B respectively.

Nice catch! Should be fixed now

>Also, with respect to tiny numbers (0.00000...), the tooltip is not currently implemented on the action page, which I think it should be?

I have only updated the number formatting in places where the `Numeral` component had been already used and this doesn't seem to be the case for the action page. I'm wondering if there's a reason behind it, do you know?

>I actually like the idea of engineering notation here, as at a glance I think it would be more informative than a string of zeroes. Then we could also have a tooltip as well, if the user wants to see the original value. I mention since you raised the question in the proposal, so I'd be interested to see how it looks.

Thanks for weighing in, since both you and Arren have proposed this I will update the formatting to use engineering notation for tiny numbers.

>And my instinct with regard to the removal of the appearance prop is that it should be left in and continued to be overriden where necessary. I say this only because I expect it was designed with that in mind originally, and follows the same custom as other core components in the dapp.

My reason for removing it was that despite being specified, it was actually pointless in all but 2 or 3 cases as it was overwritten by consuming component's styles. But I see your point, I will bring the `appearance` prop back for these 2-3 cases I mentioned

**Comment by jakubcolony** [comment]
Hey @willm30, I think I addressed all of your comments. 
As I mentioned, I brought back the `appearance` prop and it only seems relevant in one case. There was an interesting example where by setting theme to `grey`, it would only change the first letter colour, presumably to cater for this single use case in Create Payment Dialog:
image
I find changing this via the `theme` prop confusing so I left it up to the consuming component to set such styles. Let me know if you disagree.

**Review by willm30 — COMMENTED** [review]
I'm seeing the following with 999,999,999,999. Should be 999.999B?

1000b

Also not seeing the engineering notation for tiny numbers. Here I entered `0.00...1`. (can't tell you exactly as there's no way currently to see the original value). 

lowz

**Comment by willm30** [comment]
> I have only updated the number formatting in places where the `Numeral` component had been already used and this doesn't seem to be the case for the action page. I'm wondering if there's a reason behind it, do you know?

Hmm not sure. I think that would be a separate issue in any case so let's leave as is.

**Comment by jakubcolony** [comment]
Closing this as it'll now be addressed in the CDapp [issue #53](https://github.com/JoinColony/colonyCDapp/issues/53)

**Pull request closed without merging**